lib/monetary-amount.js

```diff
@@ -8,6 +11,36 @@ const _amount = new WeakMap();
 const _currencyType = new WeakMap();
 const _currencyId = new WeakMap();
 
+function convertAmountOrThrow(amount) {
+    if ((typeof amount === 'object') && (amount instanceof ethers.utils.BigNumber))
+        return amount;
+
+    if (Number.isInteger(amount))
+        return ethers.utils.bigNumberify(amount);
+
+    if ((typeof amount === 'string') && /^[-]?[0-9]+$/.test(amount))
+        return ethers.utils.bigNumberify(amount);
+
+    throw new Error('amount has invalid format');
+}
+
+function convertAddressOrThrow(ct) {
```

Comment:
LIES! This function isn't actually converting anything, it's just validating the string and ensuring it has the 0x prefix.

Reply:
He he

Reply:
Not throwing anymore, 'rectifying' :)